http_server.py

```diff
@@ -74,26 +108,56 @@ def response_path(path):
         response_path('/a_page_that_doesnt_exist.html') -> Raises a NameError
 
     """
+    content = ""
+    mime_type = ""
+
+    # We're going to exec user input - which is generally a terrible idea.
+    #   So here's a check to make sure they're not trying to kill us.
+    # valid_scripts = ['make_time.py', 'this_script_should_not_run.py']
+    valid_scripts = ["make_time.py"]
+
+    try:
+        mime_type = mimetypes.MimeTypes().guess_type("./webroot/" + path)[0]
+
+        if mime_type is None:  # Directory
+            mime_type = "text/plain"
+        mime_type = mime_type.encode()
+
+        open_path = pathlib.Path("./webroot/" + path)
+        for child in open_path.iterdir():
+            content += str(child)[8:] + "\r\n"
+        content = content.encode()
+
+    except FileNotFoundError:
+        raise NameError
+
+    except NotADirectoryError:
+        if path[-3:] == ".py" and path[1:] in valid_scripts:
+            content, mime_type = run_python_script(path)
+        else:
+            with open(("./webroot/" + path), "rb") as open_file:
+                content = open_file.read()
+
+    return content, mime_type
+
+
+def run_python_script(path):
+    """
+    Python script requested, let's run it, what could go wrong?
+    """
 
-    # TODO: Raise a NameError if the requested content is not present
-    # under webroot.
+    with stdout_io() as output:
+        exec(open("./webroot" + path).read())
```

Comment:
We only wanted to output the contents of the .py file, not run it.

Reply:
Fine as stretch goal, but honestly one should never run eval or exec in production :)

Reply:
It was the optional part of the assignment, to run that file instead of it just being returned.

Reply:
yep, aware, just wanted to make it clear that you should never do it hehe.